fsspec/spec.py

```diff
@@ -217,6 +218,13 @@ def end_transaction(self):
         """Finish write transaction, non-context version"""
         self.transaction.complete()
         self._transaction = None
+        # The invalid cache must be cleared after the transcation is completed.
+        for path in self._invalidated_caches_in_txn:
+            self.invalidate_cache(path)
+
+    def _record_invalidated_cache(self, path):
```

Comment:
Being only two lines, this could go right into `invalidate_cache` rather than make a new method

Reply:
Merged.